examples/tv-app/tv-common/clusters/channel/ChannelManager.cpp

```diff
@@ -242,10 +263,44 @@ void ChannelManager::HandleGetProgramGuide(
     // ChannelPagingStructType channelPaging;
     // channelPaging.nextToken = MakeOptional<DataModel::Nullable<Structs::PageTokenStruct::Type>>(paging);
 
-    ProgramGuideResponseType response;
-    // response.channelPagingStruct = channelPaging;
-    response.programList = DataModel::List<const ProgramType>(mPrograms.data(), mPrograms.size());
+    std::vector<ProgramType> matches;
+    for (auto const & program : mPrograms)
+    {
+        if (startTime.ValueOr(0) > program.startTime)
+        {
+            continue;
+        }
+        if (endTime.HasValue() && endTime.ValueOr(0) < program.endTime)
+        {
+            continue;
+        }
+        if (channelList.HasValue())
+        {
+            auto iter     = channelList.Value().begin();
+            bool match    = false;
+            int listCount = 0;
+            while (iter.Next() && !match)
+            {
+                listCount++;
+                auto & channel = iter.GetValue();
+                if (channel.minorNumber != program.channel.minorNumber || channel.majorNumber != program.channel.majorNumber)
+                {
+                    continue;
+                }
+                // TODO: check OTT
+                match = true;
+            }
+            if (!match && listCount > 0)
+            {
+                continue;
+            }
+        }
+        // TODO: check external id list
```

Comment:
I would prefer filing ticket for each TODO here, thanks

Reply:
This is just example code and so updated from TODO to just a comment.